test/action_controller/json/include_test.rb

```diff
@@ -38,17 +42,28 @@ def render_without_include
 
           def render_resource_with_include_hash
             setup_data
-            render json: @author, include: { posts: :comments }, adapter: :json
+            render json: @author, include: INCLUDE_HASH, adapter: :json
           end
 
           def render_resource_with_include_string
             setup_data
-            render json: @author, include: 'posts.comments', adapter: :json
+            render json: @author, include: INCLUDE_STRING, adapter: :json
           end
 
           def render_resource_with_deep_include
             setup_data
-            render json: @author, include: 'posts.comments.author', adapter: :json
+            render json: @author, include: DEEP_INCLUDE, adapter: :json
+          end
+
+          def render_without_recursive_relationships
+            # testing recursive includes ('**') can't have any cycles in the
+            # relationships, or we enter an infinite loop.
```

Comment:
This is really to support the test on line 131. If you try to use the '**' includes for the actions above, you get spec failures from infinite recursion.

I'm not sure how to resolve that without getting into the weeds. #1538 is meant to help provide some nice ways to avoid this in the presence of reference cycles in serializers, but needs to be brought up to date with master.